cluster-autoscaler/processors/processors.go

```diff
@@ -28,18 +29,23 @@ type AutoscalingProcessors struct {
 	PodListProcessor pods.PodListProcessor
 	// NodeGroupListProcessor is used to process list of NodeGroups that can be used in scale-up.
 	NodeGroupListProcessor nodegroups.NodeGroupListProcessor
+	// ScaleUpStatusProcessor is used to process the state of the cluster after a scale-up.
+	ScaleUpStatusProcessor status.ScaleUpPostProcessor
```

Comment:
I know it may be a controversial question, but is this even a 'processor' in the same sense as the above two? 
Both PodListProcessor and NodeGroupListProcessor modify the list they accept, hopefully without too many side-effects.
The one implementation for ScaleUpStatusProcessor that we have now only produces side-effects (publishing events), without modifying scale-up status. Let's say we want to add another implementation, e.g. responsible for publishing metrics related to scale-up activity. The chain processor implementation has two choices:
a) call the second one on the data returned by the former and fail if the former failed (not good for monitoring),
b) call the second one on the data returned by the former, unless the former failed, then use the original input (not good for consistency),
b) call all of them on the same input (disregarding the returned list, which may be unintuitive.)

So the question is, do we want this processor to modify the status? If yes, whichever processor is called first becomes a single point of failure for monitoring use-case (which is currently the only one.) If no, we probably shouldn't introduce this dependency of scale-up status processors on each other.

Reply:
Not sure why this comment is in this place in code (unless I misunderstand it completely). Assuming I do understand it - all of the above seem like a problem with chaining. I already commented I'd rather remove chaining until we need it. This is one more reason to do it. If we want chaining we should probably explicitly specify what changes and side effects are allowed and how we handle errors.

It's much easier to just say that if someone wants to implement a new processor it would be their job to call the default one and handle the result / errors. In this case the author of code will have full information on what every processor does and there will be no need for a generic solution.

Reply:
I suppose this comment should have been added where `ScaleUpPostProcessor` interface is defined, sorry.

Let's forget about chaining for a moment, then. We have an interface method that:
1) is supposed to return status (and context!),
2) in all places in code where it is called (excluding chaining, 3), the results of this call are disregarded (we don't even check or log error)

If we aren't considering chaining, we definitely shouldn't be returning those values. And if we were, my point is that we should probably design this interface with the intended use-case in mind.

Reply:
Oh, I didn't notice the return values. I think we shouldn't be returning any of those. Most certainly not context. Status only if we think it's meaningful to mutate it (which I don't think it is).

Reply:
Removed return values and the chaining processor.